scripts/src/javatests/org/oppia/android/scripts/maven/MavenDependenciesListWriterTest.kt

```diff
@@ -0,0 +1,338 @@
+package org.oppia.android.scripts.maven
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.oppia.android.scripts.proto.License
+import org.oppia.android.scripts.proto.MavenDependency
+import org.oppia.android.scripts.proto.MavenDependencyList
+import org.oppia.android.scripts.proto.PrimaryLinkType
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+
+private const val DATA_BINDING_POM = "https://maven.google.com/androidx/databinding/" +
+  "databinding-adapters/3.4.2/databinding-adapters-3.4.2.pom"
+private const val PROTO_LITE_POM = "https://repo1.maven.org/maven2/com/google/protobuf/" +
+  "protobuf-lite/3.0.0/protobuf-lite-3.0.0.pom"
+private const val IO_FABRIC_POM = "https://maven.google.com/io/fabric/sdk/android/" +
+  "fabric/1.4.7/fabric-1.4.7.pom"
+private const val GLIDE_ANNOTATIONS_POM = "https://repo1.maven.org/maven2/com/github/" +
+  "bumptech/glide/annotations/4.11.0/annotations-4.11.0.pom"
+private const val FIREBASE_ANALYTICS_POM = "https://maven.google.com/com/google/firebase/" +
+  "firebase-analytics/17.5.0/firebase-analytics-17.5.0.pom"
+
+/** Tests for [MavenDependenciesListWriter]. */
+class MavenDependenciesListWriterTest {
+
+  private val LICENSE_DETAILS_INCOMPLETE_FAILURE = "Licenses details are not completed."
+  private val COMPLETE_LICENSE_DETAILS_MESSAGE = "Please complete all the details" +
+    "for the following licenses:"
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  private val mockNetworkAndBazelUtils by lazy { initializeMockNetworkAndBazelUtils() }
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("scripts", "assets")
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  // 1. Some License requires manual work, script fails.
+  // 2. Some Dependency requires manual work, script fails.
+  // 3. Some License and dependency both requires manual work, script fails.
+  // 4. Textproto is complete, script passes.
+  // 5. Some dependency contains empty license list, script fails.
+  // 6. Incomplte manual work, script fails.
+  // 7. Dependencies contain invalid links, script fails.
```

Comment:
Could you add a bit more detail what are these seven points?

Reply:
These were just for my reference. Removed.